src/components/Download/DownloadOptionsForm/index.js

```diff
@@ -15,42 +15,52 @@ export const DownloadOptionsForm = ({
   dataset = null,
   buttonLabel = 'Download',
   isProcessed = false,
-  onSubmit = null
+  handleDownloadOptionsChanges = null, // for the regenerate dataset local state update
+  onSubmit = null // for the regenerate dataset download
 }) => {
   const { push } = useRouter()
   const {
-    dataset: datasetState,
-    createDataset,
+    dataset: myDataset,
     updateDataset,
-    regeneratedDataset,
     getDownloadOptions,
     updateDownloadOptions
   } = useDatasetManager()
   const { setResponsive } = useResponsive()
-  const selectedDataset = dataset || datasetState
+
+  const [currentDataset, setCurrentDataset] = useState(null)
```

Comment:
I don't think this component needs to infer which dataset should be modified here. Can we have the dataset that is passed in be the one that is being modified always? So if we want to handle myDataset we pass that in and if it is a regenerated dataset we do that?

Reply:
I've removed the local state `currentDataset` and made the `dataset` prop mandatory for this `DownloadOptionsForm` component, and adjusted the codebase accordingly 👍